PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -58,25 +58,30 @@ def __mnemonic2entropy(mnemonic: str) -> bytes:
     pass
 
 
-def __is_valid_entropy(entropy: bytes) -> bool:
+def is_valid_entropy(entropy: bytes) -> bool:
```

Comment:
so these should be public now?

Reply:
In case of invalid input data, entropy would be passed from CLI tool to `generate`.

https://github.com/mvondracek/PA193_mnemonic_Slytherin/blob/5f7951ec80a9112d8ac91c6b111690f77d8213fc/PA193_mnemonic_Slytherin/mnemoniccli.py#L114

According to current package logic, `generate` would raise `ValueError` on invalid entropy.
https://github.com/mvondracek/PA193_mnemonic_Slytherin/blob/5f7951ec80a9112d8ac91c6b111690f77d8213fc/PA193_mnemonic_Slytherin/mnemonic.py#L95-L104

I could then catch `ValueError` from `generate` in CLI tool. However, how can I be sure that it's result of invalid entropy and that it is not raised by some other error in the rest of `generate`?
https://github.com/mvondracek/PA193_mnemonic_Slytherin/blob/5f7951ec80a9112d8ac91c6b111690f77d8213fc/PA193_mnemonic_Slytherin/mnemonic.py#L106-L107

We could inherit from `ValueError` our exception let's say `InvalidEntropyError`, which would be raised by `generate` in this situation. CLI tool could catch it and know what's the cause of this error *and*  it would maintain semantic of `ValueError` as invalid parameter.

I wanted *not to* alter logic of the package too much while I work on CLI tool in this PR. This PR is about CLI tool.

So for this PR, these functions became public and the CLI tool performs LBYL.
https://github.com/mvondracek/PA193_mnemonic_Slytherin/blob/5f7951ec80a9112d8ac91c6b111690f77d8213fc/PA193_mnemonic_Slytherin/mnemoniccli.py#L106-L113

I see it as a next step to consider EAFP with raised `InvalidEntropyError` or even better with class for `Entropy` which ensures that only valid entropy objects can be instantiated. This way our core functions would just check for instance of `Entropy` (`Entropy` and its subclasses). If we would like to use power of duck typing, we would not check for instance of `Entropy`, but for existence of methods we would like to use.